core/src/main/java/org/apache/iceberg/BaseCombinedScanTask.java

```diff
@@ -19,21 +19,22 @@
 
 package org.apache.iceberg;
 
+import java.util.Arrays;
 import java.util.Collection;
 import java.util.List;
 import org.apache.iceberg.relocated.com.google.common.base.Joiner;
 import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
-import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
 
 public class BaseCombinedScanTask implements CombinedScanTask {
   private final List<FileScanTask> tasks;
 
   public BaseCombinedScanTask(FileScanTask... tasks) {
-    this.tasks = ImmutableList.copyOf(tasks);
+    this.tasks = Arrays.asList(tasks);
   }
 
   public BaseCombinedScanTask(List<FileScanTask> tasks) {
-    this.tasks = ImmutableList.copyOf(tasks);
+    this.tasks = copyList(tasks);
```

Comment:
What about a slightly different approach?

Right now, the main problem is that `List` can't be easily serialized. But we know that `tasks` are `FileScanTask` and never changes, so we could use `private final FileScanTask[] tasks` instead, which had no problems with serialization.

Then this would just need to copy into the array, `tasks.stream().toArray(FileScanTask[]::new)`, and we could update `files` to return `ImmutableList.copyOf(tasks)`.

By making those changes, we still end up with an immutable list returned by `files`, but take care of the serialization issue and don't need to add `copyList`.

Reply:
Ah OK that also makes sense - I realized we'd probably want to make sure `tasks` cannot be modified but `files()` opens it for modification outside of this class.

I simply thought about defining `tasks` as `ArrayList<FileScanTask>` which also ensures no problem with serialization, but the efforts would be similar (had to make change on constructors and `files()`) so either is fine for me. I'll go through changing it to array first. Thanks!